### PR TITLE
Generate and use enum types in parameters for typescript-axios

### LINKED ISSUE #13439: [REQ] Use generated enums in operations for typescript-axios
### Is your feature request related to a problem? Please describe.

When using API functions generated from this, I want to be able to also use the enum types. The enums are generated for models, but they should also be generated for operations too.

## Describe the solution you'd like

Generate and use enum types in operations.

## Describe alternatives you've considered

Just hard-coding the enum values when using operations, but that's not great.

## Additional context

e.g.
```ts
findPetsByStatus: async (status: Array<'available' | 'pending' | 'sold'>, ...
```
should be
```ts
findPetsByStatus: async (status: Array<FindPetsByStatusStatusEnum>, ...
...
export const FindPetsByStatusStatusEnum = {
    Available: 'available',
    Pending: 'pending',
    Sold: 'sold'
} as const;
export type FindPetsByStatusStatusEnum = typeof FindPetsByStatusStatusEnum[keyof typeof FindPetsByStatusStatusEnum];
```

### PR DESCRIPTION
**Pull request by chaayac**
Resolves #13439

Changes enum implementation for operators in the typescript-axios generator to more closely follow the fetch generator - generating actual types for enums defined in operators rather than using union (`|`) syntax.

This should be a non-breaking change since the effective types are the same, but please correct me if i'm wrong.

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (6.1.0) (minor release - breaking changes with fallbacks), `7.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.
@TiFu (2017/07) @taxpon (2017/07) @sebastianhaas (2017/07) @kenisteward (2017/07) @Vrolijkx (2017/09) @macjohnny (2018/01) @topce (2018/10) @akehir (2019/07) @petejohansonxo (2019/11) @amakhrov (2020/02) @davidgamero (2022/03) @mkusaka (2022/04)

### CODE REVIEW

**Comment by galihputera** [comment]
up

**Comment by amakhrov** [comment]
> This should be a non-breaking change since the effective types are the same

If real enums are used (via the `stringEnums` option), it will be a breaking change, since strings are not assignable to enums
Screen Shot 2022-09-19 at 10 24 16 AM

**Comment by chaayac** [comment]
@amakhrov I see, any suggestions to fix that, or should I just merge to 7.0.x?

**Comment by amakhrov** [comment]
I think targeting this for the 7.x branch is reasonable

**Comment by chaayac** [comment]
@amakhrov I switched to merge to 7.0.x.

**Comment by chaayac** [comment]
@amakhrov can I get your review/approval?

I'm not sure what the failing bitrise test is, could use some help with that too please.

File: modules/openapi-generator/src/main/resources/typescript-axios/apiInner.mustache
Comment (amakhrov) [inline modules/openapi-generator/src/main/resources/typescript-axios/apiInner.mustache:372]: **nitpick:** this seems redundant - since you render output for each collection element matching the filter (`{{#allParams}} {{#isEnum}}`). If the isEnum condition is never satisfied, nothing is rendered, even without the top-level wrapping hasEnum check

File: modules/openapi-generator/src/main/resources/typescript-axios/apiInner.mustache
Comment (amakhrov) [inline modules/openapi-generator/src/main/resources/typescript-axios/apiInner.mustache:403]: The block the generates different enum types duplicates existing enum template defined in `modelEnum.mustache`. Suggest reusing that - will make it easier to maintain going forward.

The main difference is that  `modelEnum.mustache` uses the `classname` property of the enum, not `enumName`. So in the Java code you might need to update that to match your desired naming convention (prefix with the operation name)

Comment (chaayac) [reply modules/openapi-generator/src/main/resources/typescript-axios/apiInner.mustache:403]: `CodegenParameter`s don't have a `classname` property (it's only in `CodegenModel`s), which is the reason I opted for doing it this way (also because it's the way the fetch generator does it too).

File: modules/openapi-generator/src/main/resources/typescript-axios/apiInner.mustache
Comment (chaayac) [reply modules/openapi-generator/src/main/resources/typescript-axios/apiInner.mustache]: removed these.

File: modules/openapi-generator/src/main/resources/typescript-axios/apiInner.mustache
Comment (chaayac) [reply modules/openapi-generator/src/main/resources/typescript-axios/apiInner.mustache:402]: @amakhrov any suggestion on how to go about this? should i make another template for params, or make it conditional (like use `classname` if exists, else use `enumName` but that seems kind of hacky?)

Comment (amakhrov) [reply modules/openapi-generator/src/main/resources/typescript-axios/apiInner.mustache:402]: This is unfortunate.
Looks like separate implementation for CodegenParameter, as you have in your PR is the best way forward right now.

**Review by amakhrov — APPROVED** [review]
Looks good to me, thanks for your contribution!

**Comment by chaayac** [comment]
Thanks for the review. Is there anything else I need to do to get this merged? Need someone with write access to do it please.

@amakhrov or @jimschubert?

**Comment by amakhrov** [comment]
One of the tests failed in CI.
Looks to be what has just been fixed in this PR: https://github.com/OpenAPITools/openapi-generator/pull/13526 . Perhaps need to merge in latest changes from `master`  here?

**Comment by chaayac** [comment]
I updated the PR with the fix. Can't directly merge because the fix isn't on the `7.0.x` branch.

**Comment by chaayac** [comment]
Bumping this for merging please @amakhrov.

**Comment by amakhrov** [comment]
@macjohnny

**Pull request merged**
